test/unit/UpdatesPublisherTest.js

```diff
@@ -0,0 +1,114 @@
+require('require-self-ref')
+const test = require('ava')
+const sinon = require('sinon')
+const proxyquire = require('proxyquire')
+const MockProducer = require('~/test/mocks/mockProducer')
+const MockChangesStream = require('~/test/mocks/mockChangesStream')
+const Promise = require('bluebird')
+proxyquire.noPreserveCache()
+
+function makeProducerAndStubs (sandbox) {
+  const createProducerStub = sandbox.stub()
+  const publisher = proxyquire('~/src/UpdatesPublisher', {
+    'windbreaker-service-util/queue': {
+      createProducer: createProducerStub
+    }
+  })
+  return {
+    createProducerStub: createProducerStub,
+    publisher: publisher
+  }
+}
+
+test.beforeEach('setup mock channel and connections', (t) => {
+  const producerOptions = {
+    queueName: 'test-queue'
+  }
+
+  const sandbox = sinon.sandbox.create()
+  const {publisher, createProducerStub} = makeProducerAndStubs(sandbox)
+
+  t.context = {
+    producerOptions,
+    sandbox,
+    publisher,
+    createProducerStub
+  }
+})
+
+test.afterEach('clean up', (t) => {
+  const { sandbox } = t.context
+
+  sandbox.restore()
+})
+
+test.serial('#setupProducer', async (t) => {
+  const {producerOptions, createProducerStub, publisher} = t.context
+  publisher.configure(producerOptions, 'http://unhappypath.sad', 'http://unhappierpath.sad', console)
+  createProducerStub.onFirstCall().returns(Promise.reject(new Error('expected rejection')))
+  createProducerStub.onSecondCall().returns(Promise.resolve())
+  await publisher.setupProducer()
+  t.true(createProducerStub.calledTwice)
+})
+test.serial('#setupChanges', async (t) => {
+  const {producerOptions, sandbox, publisher} = t.context
+  publisher.configure(producerOptions, 'http://unhappypath.sad', 'http://unhappierpath.sad', console)
+  const consoleSpy = sandbox.spy(console, 'info')
+  await publisher.setupChanges()
+  t.true(consoleSpy.calledWith('changes stream successfully created'))
+  publisher.changes.destroy()
+})
+test.serial('#start', async (t) => {
```

Comment:
I like that you are specifying which function is being tested, but you should be adding more meaningful test descriptions. Exactly what is being tested here other than just the `start` function? What part of `start` is being tested? All of it? Also in what way is it significant? (see my comment below)